resources/aws/policy.go

```diff
@@ -141,7 +143,16 @@ func (p *Policy) clusterRoleName() string {
 }
 
 func (p *Policy) CreateIfNotExists() (bool, error) {
-	return false, fmt.Errorf("instance profiles cannot be reused")
+	err := p.CreateOrFail()
+	if err != nil {
+		if strings.Contains(err.Error(), awsclient.RoleDuplicate) {
+			return false, nil
+		}
```

Comment:
Sadly this string matching is needed to check for the error.

Reply:
It would be nice to keep a flat condition flow and to use a custom error matcher. 

```
if IsIAMRoleDuplicateError(err) {
    //
} else if err != nil {
    // 
}
```

Reply:
Done.